tests/test-files.c

```diff
@@ -251,6 +251,51 @@ START_TEST(test_path_relative_child_01)
 END_TEST
 
 
+void
+test_set_absolute(const char *p, const char *expected)
+{
+    struct cork_path  *actual;
+
+    fprintf(stderr, "set_absolute(\"%s\") ?= \"%s\"\n",
+            (p == NULL)? "": p,
+            expected);
+
+    actual = cork_path_new(p);
+    cork_path_set_absolute(actual);
+    if ((p != NULL) && (p[0] == '/')) {
+        /* If the first char in p is a '/', then we want to
+         * test that already have an absolute path string. */
+        verify_path_content(actual, expected);
+    } else {
+        /* Otherwise, we have to construct a new expected path
+         * string using cork_path_cwd since the prepended $ROOT
+         * path is unknown. */
+        struct cork_path  *root_path;
+        struct cork_buffer  *expected_path = cork_buffer_new();
+        root_path = cork_path_cwd();
```

Comment:
Can you run this through valgrind real quick?  I think you need to free `root_path` as well.